src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListenerImplBase.cs

```diff
@@ -98,6 +98,12 @@ public void OnNext(KeyValuePair<string, object> kv)
 				return;
 			}
 
+			if (_agent.Tracer.CurrentTransaction == null)
```

Comment:
As I see, `ProcessStartEvent` already has such a check.

https://github.com/elastic/apm-agent-dotnet/blob/460b0078f3d56f677d194c5cb650293dac7ba2db/src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListenerImplBase.cs#L121-L126

So, it means that your change will affect Stop and Exception events. If it's possible that the current transaction is null during the HTTP start event (this is mentioned in issue), I think it can be possible that the current transaction isn't null during the start event but will be null during the stop event?

Reply:
Interesting point. 

> I think it can be possible that the current transaction isn't null during the start event but will be null during the stop event.

Since the current transaction is in the async local storage, this would sounds like a scenario where the `HttpClient` class has a bug - otherwise it'd be impossible to stop a transaction in the middle of an http call. Or is there any way to execute code on the given Task during a method call of the `HttpClient`?

Nevertheless this may indeed happen - technically it's definitely possible - I agree on that.

I change the code - now all the changes are in `ProcessStopEvent` - if the transaction is null, then we just print a debug log.